src/ui.py

```diff
@@ -33,14 +33,15 @@ def on_figure_style_change(_a, _b, self):
 
 
 def on_items_change(data, _ignored, self):
+    data = self.get_data()
     item_list = self.get_window().get_item_list()
     while item_list.get_last_child() is not None:
         item_list.remove(item_list.get_last_child())
 
-    for item in self.get_data():
-        item_list.append(ItemBox(self, item))
+    for count, item in enumerate(data):
+        item_list.append(ItemBox(self, item, count))
```

Comment:
The `count` variable sets the index for the new item to be added, right? Wouldn't it make sense to call it index then?